python/lsst/ts/standardscripts/maintel/setup_mtcs.py

```diff
@@ -0,0 +1,380 @@
+            type: object
+            additionalProperties: false
+            properties:
+              mtptg: 
+                description: Override configuration for MTPtg.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              mtmount: 
+                description: Override configuration for MTMount.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              mtrotator: 
+                description: Override configuration for MTRotator.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              mtm1m3: 
+                description: Override configuration for MTM1M3.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: Default
+              mtm2:
+                description: Override configuration for MTM2.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              mthexapod_1:
+                description: Override configuration for Camera Hexapod.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+              mthexapod_2:
+                description: Override configuration for M2 Hexapod.
+                anyOf:
+                  - type: string
+                  - type: "null"
+                default: null
+        """
+        return yaml.safe_load(schema)
+
+    async def configure(self, config):
+        """Handle script input configuration.
+
+        Parameters
+        ----------
+        config: `types.SimpleNamespace`
+          Configuration data. See `get_schema` for information about data
+          structure.
+        """
+        self.log.debug(f"Enable CCW following: {config.ccw_following}")
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+        self.activity_estimated_duration = dict(
+            enable_components=dict(time=5.0, repeats=8),  # Enable components
+            move_to_zenith=dict(time=10.0, repeats=1),  # Move mount to zenith
+            raise_m1m3=dict(time=600.0, repeats=1),  # Raise m1m3
+            enable_corrections=dict(time=5.0, repeats=4),  # Enable/reset corrections
+            get_telemetry=dict(time=5.0, repeats=6),  # Timeout extracting telemetries
+        )
+
+        metadata.duration = sum(
+            [
+                self.activity_estimated_duration[action]["time"]
+                * self.activity_estimated_duration[action]["repeats"]
+                for action in self.activity_estimated_duration
+            ]
+        )
+
+    async def run(self):
+        await self.checkpoint("Check that MTCS Components have heartbeats")
```

Comment:
I liked how this was done. One thing that I'd suggest though, is to put all there checkpoints/methods pairs in a list of tuples, then you can use a `for` loop here to execute them.. 

For example, in the `__init__` method you would do:

```
self.checkpoints_activities = [ 
    ("Check that MTCS Components have heartbeats", self.mtcs.assert_liveliness),
    ("Start MTPtg", self.start_mtptg),
    ...
]
```

Then here you would do:

```
for checkpoint, activity in self.checkpoints_activities:
    await self.checkpoint(checkpoint)
    await activity()
```

Reply:
done